Python/traceback.c

```diff
@@ -708,15 +740,17 @@ tb_displayline(PyTracebackObject* tb, PyObject *f, PyObject *filename, int linen
                                 filename, lineno, name);
     if (line == NULL)
         return -1;
-    err = PyFile_WriteObject(line, f, Py_PRINT_RAW);
+    err = _Py_WriteIndentedMargin(margin_indent, margin, f);
+    err |= PyFile_WriteObject(line, f, Py_PRINT_RAW);
     Py_DECREF(line);
     if (err != 0)
         return err;
 
     int truncation = _TRACEBACK_SOURCE_LINE_INDENT;
     PyObject* source_line = NULL;
-    if (_Py_DisplaySourceLine(f, filename, lineno, _TRACEBACK_SOURCE_LINE_INDENT,
-                               &truncation, &source_line) != 0 || !source_line) {
+    if (display_source_line_with_margin(
+            f, filename, lineno, _TRACEBACK_SOURCE_LINE_INDENT,
+            margin_indent, margin, &truncation, &source_line) != 0 || !source_line) {
```

Comment:
AFAICS, this branch is not exercised in the test suite.

Also, I'd consider refactoring the condition for the sake of readability (a three-line `if` condition is IMO hard on the eyes). For example:

```suggestion
    int rc = display_source_line_with_margin(f, filename, lineno,                                                     
                                             _TRACEBACK_SOURCE_LINE_INDENT,                                           
                                             margin_indent, margin, &truncation                                       
                                             &source_line);                                                           
    if (rc < 0 || !source_line) {  
```

Reply:
Do you mean the branch where display_source_line_with_margin fails?

Reply:
I mean the branch where this `if` statement evaluates to true. That is, either `rc < 0` or `source_line == 0`.

Reply:
I'm not sure how to cause that in a test (anyway it's not new to this PR, right?)